plotly/tools.py

```diff
@@ -2871,3 +2934,188 @@ def get_candle_decrease(self):
         return (decrease_x, decrease_close, decrease_dif,
                 stick_decrease_y, stick_decrease_x)
 
+class _Dendrogram(FigureFactory):
+    
+    """
+    Refer to FigureFactory.create_dendrogram() for docstring.
+    """
+
+    def __init__(self, X, orientation='bottom', labels=None, colorscale=None, \
+                 width="100%", height="100%", xaxis='xaxis', yaxis='yaxis' ):
+        self.orientation = orientation
+        self.labels = labels
+        self.xaxis = xaxis
+        self.yaxis = yaxis
+        self.data = []
+        self.leaves = []
+        self.sign = {self.xaxis: 1, self.yaxis: 1}
+        self.layout = {self.xaxis: {}, self.yaxis: {}}
+
+        if self.orientation in ['left', 'bottom']:
+            self.sign[self.xaxis] = 1
+        else:
+            self.sign[self.xaxis] = -1
+
+        if self.orientation in ['right', 'bottom']:
+            self.sign[self.yaxis] = 1
+        else:
+            self.sign[self.yaxis] = -1
```

Comment:
Apologies, I think I may have just read over that too fast, looks good. Also, these are really great plots!!

I wouldn't worry about that for now, let's get this merged in! I'll be more present so that we can get this in soon so we don't keep leaving you hanging.